src/messages/search/MessageFlagsPredicate.cpp

```diff
@@ -0,0 +1,39 @@
+#include "messages/search/MessageFlagsPredicate.hpp"
+
+namespace chatterino {
+
+MessageFlagsPredicate::MessageFlagsPredicate(const QString &flags)
+    : flags_()
+{
+    // Check if any comma-seperated values were passed and transform those
+    for (const auto &flag : flags.split(',', QString::SkipEmptyParts))
+    {
+        if (flag == "deleted" || flag == "disabled")
+        {
+            this->flags_.set(MessageFlag::Disabled);
+        }
+        else if (flag == "sub" || flag == "subscription")
+        {
+            this->flags_.set(MessageFlag::Subscription);
+        }
+        else if (flag == "timeout")
```

Comment:
Could also include ban there, since it also works for ban messages

```suggestion
        else if (flag == "timeout" || flag == "ban")
```